cmd/etcdbackup/etcd_backup.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"time"
+
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/coreos/etcd/pkg/transport"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/log"
+)
+
+const (
+	cMaxBackups = 6
+)
+
+var (
+	logLevel    = flag.String("loglevel", "Debug", "Valid values are Debug, Info, Warning, Error")
+	blobName    = flag.String("blobname", "", "Name of the blob (without the container)")
+	group       = flag.String("group", "hourly", "The Blob will be called etcd/backup-<group>-<timestamp>")
+	destination = flag.String("destination", "", "Where to place the blob on the filesystem")
+	gitCommit   = "unknown"
+)
+
+type backupCmd struct {
+	abs           *azstorage.BlobStorageClient
+	endpoints     []string
+	etcdTLSConfig *tls.Config
+}
+
+// saveSnap uses backup writer to save etcd snapshot to a specified path
+// and returns backup etcd server's kv store revision and its version.
+func (b *backupCmd) saveSnap(ctx context.Context, blobName string) (int64, string, error) {
+	etcdcli, rev, err := b.etcdClientWithMaxRevision(ctx)
+	if err != nil {
+		return 0, "", fmt.Errorf("create etcd client failed: %v", err)
+	}
+	defer etcdcli.Close()
+
+	resp, err := etcdcli.Status(ctx, etcdcli.Endpoints()[0])
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to retrieve etcd version from the status call: %v", err)
+	}
+
+	rc, err := etcdcli.Snapshot(ctx)
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to receive snapshot (%v)", err)
+	}
+	defer rc.Close()
+
+	bw, err := newBlobWriter(b.abs, blobName)
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to create blob (%v)", err)
+	}
+	defer bw.Close()
```

Comment:
agreed not to use defer here as it's critical (presumably a mock unit test will be able to confirm this - open follow up if not implemented here)

Reply:
done